src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -329,6 +329,31 @@ var _ = Describe("installer HostRoleMaster role", func() {
 			Expect(exit).Should(Equal(false))
 		})
 
+		It("WaitAndUpdateNodesStatus including joined state from reboot", func() {
+			joined := []models.HostStage{models.HostStageJoined,
+				models.HostStageJoined,
+				models.HostStageJoined}
+
+			hosts := create3Hosts(models.HostStatusInstalling, models.HostStageRebooting, "")
```

Comment:
It's not clear to me why a PR that changes the behavior in the case of SNO is adding this single test with 3 nodes.
Though it shouldn't happen, I guess it is valid, yet please add a test for the actual flow this code tries to solve: SNO transitioning to `Joined` from `Rebooting`

Reply:
cause it is not specific for sno, this change allows transition from reboot to joined.